File: Golang/Longest_Common_Prefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    
    prefix := strs[0]
    
    for i := 1; i < len(strs); i++ {
        for j := 0; j < len(prefix); j++ {
            if j >= len(strs[i]) || prefix[j] != strs[i][j] {
                prefix = prefix[:j]
                break
            }
        }
        if prefix == "" {
            break
        }
    }
    return prefix
}

File: Kotlin/Longest_Common_Prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        if (strs.isEmpty()) return ""
        
        var prefix = strs[0]
        for (i in 1 until strs.size) {
            var j = 0
            while (j < prefix.length && j < strs[i].length && prefix[j] == strs[i][j]) {
                j++
            }
            prefix = prefix.substring(0, j)
            if (prefix == "") return ""
        }
        return prefix
    }
}
